### PR TITLE
Route whitelisting

### PR DESCRIPTION
**Pull request by RenaudLN**
Allow users to whitelist routes:
* Routes should follow the Flask syntax
* Due to Dash's internals, a number of base routes need to be whitelisted as well (js dependencies, layout route...)
* Due to Dash's internals, whitelisting Flask routes is not enough: the `_dash-update-component` route responsible for browser navigation needs to be whitelisted for the relevant routes.

The PR shows an example of how to use the feature in the updated Readme.

### CODE REVIEW

**Comment by RenaudLN** [comment]
Hey @alexcjohnson, should be ready to go!

File: dash_auth/public_routes.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+]
+PUBLIC_ROUTES = "PUBLIC_ROUTES"
+PUBLIC_CALLBACKS = "PUBLIC_CALLBACKS"
+
+default_config = Flask.default_config
+Flask.default_config = ImmutableDict(
+    **default_config,
+    **{
+        PUBLIC_ROUTES: Map([]).bind(""),
+        PUBLIC_CALLBACKS: [],
```
Comment (alexcjohnson) [inline dash_auth/public_routes.py:34]: Do we need to touch `Flask.default_config`? Feels like `PUBLIC_ROUTES` could go on the app itself (`app._PUBLIC_ROUTES` perhaps?) and `PUBLIC_CALLBACKS` could just be a module-level variable here, and that would keep our impact a bit more contained. Would that work?

Comment (RenaudLN) [reply dash_auth/public_routes.py:34]: Are you happy for me to set new attributes on the `Dash` instance like this?

Another option would be to not set default values and check their existence in the server config before using them. I think I'd go for that option if we want to have less impact.

Comment (alexcjohnson) [reply dash_auth/public_routes.py]: Excellent, I like that solution a lot!

**Review by alexcjohnson — APPROVED** [review]
💃 Great! I'll merge and create v2.2.0 🎉

**Comment by alexcjohnson** [comment]
v2.2.0 is released. Thanks for the contribution @RenaudLN!
https://pypi.org/project/dash-auth/
https://github.com/plotly/dash-auth/releases/tag/v2.2.0

**Pull request merged**